hack/generator/pkg/astmodel/type_visitor.go

```diff
@@ -128,23 +128,29 @@ func IdentityVisitOfPrimitiveType(_ *TypeVisitor, it *PrimitiveType, _ interface
 	return it, nil
 }
 
-func identityVisitObjectTypePerPropertyContext(_ *PropertyDefinition, ctx interface{}) interface{} {
-	return ctx
+func identityVisitObjectTypePerPropertyContext(_ *ObjectType, _ *PropertyDefinition, ctx interface{}) (interface{}, error) {
+	return ctx, nil
 }
 
 var IdentityVisitOfObjectType = MakeIdentityVisitOfObjectTypeWithPerPropertyContext(identityVisitObjectTypePerPropertyContext)
 
-// MakeIdentityVisitOfObjectTypeWithPerPropertyContext creates an visitor function which creates a per-property context before visiting each
+// MakeIdentityVisitOfObjectTypeWithPerPropertyContext creates a visitor function which creates a per-property context before visiting each
 // property of the ObjectType
 func MakeIdentityVisitOfObjectTypeWithPerPropertyContext(
```

Comment:
This name is too wordy - it would be better to make this just `MakeIdentityVisitOfObjectType`. Since there's no way to call it without passing a `makeCtx` func, and there's no other simpler `MakeIdentityVisitOfObjectType` in the namespace that it needs to be distinguished from, it would be as clear without the extra words in the name. 

Basically the name should say what the function does (makes an identity object type visitor), not how it does it (by making a new context for every property).

Reply:
> Basically the name should say what the function does (makes an identity object type visit func), not how it does it (by making a new context for every property).

I sometimes struggle with this =)

Originally I was thinking that I really wanted to highlight the difference between `MakeIdentityVisitOfObjectTypeWithPerPropertyContext` and `IdentityVisitOfObjectType`, but you're right that in practice I'm not sure that distinction helps all that much.

I've done this.